spec/requests/api/logging_spec.rb

```diff
@@ -0,0 +1,51 @@
+#
+# REST API Logging Tests
+#
+describe ApiController do
+  describe "Successful Requests logging" do
+    LOG_REQUESTS = {
+      "API Request"    => nil,
+      "Authentication" => nil,
+      "Authorization"  => nil,
+      "Request"        => nil,
+      "Parameters"     => nil,
+      "Response"       => nil
+    }.freeze
+
+    def expect_log_requests(expectations)
+      expectations.each do |category, expectation|
+        expect_any_instance_of(ApiController).to receive(:log_request)
+          .with(category, expectation ? expectation : kind_of(Hash))
```

Comment:
Ah, I see. I misread that. Maybe move the `kind_of(Hash)` to the `LOG_REQUESTS`. Since it's a constant that probably won't work, so it may work better as a method?